lib/item_repo.rb

```diff
@@ -33,34 +33,25 @@ def find_by_name(name)
 
   def find_all_with_description(description)
     @items.values.find_all do |item|
-      item.name.downcase.include? name.downcase
+      item.description.downcase.include? description.downcase
     end
   end
 
   def find_all_by_price(price)
-    @items.values.reduce([]) do |result, item|
-      result << item if item.unit_price.to_i == price
-      result
+    @items.values.find_all do |item|
+      item.unit_price.to_s.include? price.to_s
     end
   end
 
-  def find_all_by_price_in_range(low, high)
-    @items.reduce([]) do |result, item|
-      if item.unit_price.include?(low.to_i..high.to_i)
-        result << merchant
-      else
-        result
-      end
+  def find_all_by_price_in_range(range)
+      @items.values.find_all do |item|
+        range.include? item.unit_price.to_s
     end
   end
 
   def find_all_by_merchant_id(merchant_id)
-    @items.reduce([]) do |result, item|
-      if item.merchant_id == merchant_id
-        result << merchant
-      else
-        result
-      end
+    @items.values.find_all do |item|
+      item.merchant_id == merchant_id
```

Comment:
I like this! Looks cleaner than what we had before and does the exact same thing. Perfect.